tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -371,6 +348,8 @@ std::vector<TensorView*> mlp_backwards(
 } // namespace
 
 TEST_P(DistributedTransformerTest, MLP_Layer) {
+  preseg_passes::OptimizationPassGuard<preseg_passes::PropagateShardingsPass>
```

Comment:
Why? We'd better test the real behavior without turning any knobs. I guess in this case PropagateShardingPass failed to process MatmulOp? If so, can you file a bug and `TODO(bug number)`?

Reply:
Since the manual sharding propagations covered all the tensors, it's actually fine to let the current pass run since it will do nothing. I left it more as a sanity check that the manual propagations were doing what I thought.

Reply:
Got it! I'd let the pass run because it's closer to the real integration. The user (likely Thunder) is expected to annotate shardings but not expected to turn off individual passes.